src/server/journal/streamer.cc

```diff
@@ -211,6 +212,8 @@ void RestoreStreamer::Start(util::FiberSocketBase* dest, bool send_lsn) {
 
     bool written = false;
     cursor = pt->Traverse(cursor, [&](PrimeTable::bucket_iterator it) {
+      BucketSerializationGuard guard(&bucket_ser_);
```

Comment:
How about adding this to `Traverse()` and `OnDbChange` so that it will be impossible to forget to lock them?

Reply:
`Traverse()` is a member of dashtable. The lock is associated with the callback that we pass when we call it. As for `OnDbChange` I already added it ?? Maybe I misinterpreted something in your comment.